File: 프로그래머스/연습문제/12917_문자열_내림차순으로_배치하기.java
//https://school.programmers.co.kr/learn/courses/30/lessons/12917

// Todo: String.toCharArray(), sort(Comparator.reverseOrder()), StringBuilder()

import java.util.*;

class Solution {
    public String solution(String s) {
        List<Character> list = new ArrayList<>();

        for (char c : s.toCharArray()) {
            list.add(c);
        }

        list.sort(Comparator.reverseOrder());

        StringBuilder answer = new StringBuilder();
        for (char c : list) {
            answer.append(c);
        }

        return answer.toString();
    }
}

File: 프로그래머스/연습문제/12917_문자열_내림차순으로_배치하기.cpp
//https://school.programmers.co.kr/learn/courses/30/lessons/12917

#include <string>
#include <vector>
#include <algorithm>

using namespace std;

string solution(string s) {
    string answer = "";
    vector<char> vec;

    for(auto ss : s) {
        vec.push_back(ss);
    }

    sort(vec.begin(), vec.end(), greater<>());

    for(auto v : vec) {
        answer += v;
    }

    return answer;
}